firebase_test.go

```diff
@@ -338,6 +338,224 @@ func TestVersion(t *testing.T) {
 	}
 }
 
+func TestAutoInitEnv(t *testing.T) {
+	configOld, configExists := overwriteEnv(FirebaseEnvName, "testdata/firebase_config.json", t)
+	defer reinstateEnv(FirebaseEnvName, configOld, configExists)
+
+	varName := "GOOGLE_APPLICATION_CREDENTIALS"
+	credOld, credExists := overwriteEnv(varName, "testdata/service_account.json", t)
+	defer reinstateEnv(varName, credOld, credExists)
+
+	app, err := NewApp(context.Background(), nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	want := &Config{
+		DatabaseURL:   "https://hipster-chat.firebaseio.mock",
+		ProjectID:     "hipster-chat-mock",
+		StorageBucket: "hipster-chat.appspot.mock",
+	}
+	compareConfig(app, want, t)
+}
+
+func TestAutoInitNoEnvVar(t *testing.T) {
+	configOld, configExists := overwriteEnv(FirebaseEnvName, "", t)
+	defer reinstateEnv(FirebaseEnvName, configOld, configExists)
+
+	err := os.Unsetenv(FirebaseEnvName)
```

Comment:
If overwriteEnv treats "" as unset (as commented below), then this code can go away.

Reply:
Iv'e change that but I think there should be a distinct test case for this